src/index.js

```diff
@@ -1,27 +1,42 @@
 // @flow
 
 // Helpers
-export function createToken(name: string): any {
-  // $FlowFixMe
-  return () => {
-    throw new Error(`Missing required value for token: ${name}.`);
-  };
+export const TokenType = {
+  Required: 0,
+  Optional: 1,
+};
+function Ref() {}
+class TokenImpl {
+  name: string;
+  ref: mixed;
+  type: $Values<typeof TokenType>;
+
+  constructor(name: string, ref: mixed) {
+    this.name = name;
+    this.ref = ref || new Ref();
+    this.type = ref ? TokenType.Optional : TokenType.Required;
+    if (!ref) {
+      // $FlowFixMe
+      this.optional = new TokenImpl(name, this.ref);
+    }
+  }
 }
 
-export function createOptionalToken<Default>(
-  name: string,
-  defaultValue: Default
-): Default {
+export type Token<T> = {
```

Comment:
This will be the interface exported and consumed by folks interacting with `Tokens`.  By default, they are required, but a `.optional` property allows a plugin author to designate a dependency as optional.